File: BackEnd/src/main/java/com/example/backend/dtos/ChallengeDisplayDTO.java
package com.example.backend.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChallengeDisplayDTO {
    private Long id;
    private String title;
    private int numQuestions;
    private String status;
    private String creationDate;
    private int numberOfUses;
    private String templateName;
}

File: projectAngular/src/app/entities/ChallengeDisplayDTO.ts
export interface ChallengeDisplayDTO {
    id: number;
    title: string;
    numQuestions: number;
    status: string;          // e.g., 'PUBLISHED', 'DRAFT', etc.
    creationDate: string;    // Format ISO string, ex: '2025-07-09T12:34:56'
    numberOfUses: number;
    templateName: string;
}
